analyzers/src/SonarAnalyzer.CSharp/Rules/DatabasePasswordsShouldBeSecure.cs

```diff
@@ -129,6 +129,11 @@ private void ReportEmptyPassword(XDocument doc, string webConfigPath, Compilatio
 
         private void ReportEmptyPassword(JsonNode doc, string appSettingsPath, CompilationAnalysisContext c)
         {
+            if (!doc.ContainsKey("ConnectionStrings"))
+            {
+                return;
+            }
+
             foreach (var key in doc["ConnectionStrings"].Keys)
             {
                 if (doc["ConnectionStrings"][key] is var connectionStringNode && IsVulnerable(connectionStringNode.Value.ToString()))
```

Comment:
`connectionStringNode.Value` might throw too. You may also add a test case.
```suggestion
                if (doc["ConnectionStrings"][key] is { Kind: Kind.Object } connectionStringNode && IsVulnerable(connectionStringNode.Value.ToString()))
```

Reply:
It is not possible to have at line 139 something that is not of `Kind.Object` as there will be an exception thrown on line 134 in case the kind of the node is not Object.

Reply:
While most of these examples are not valid according to the appsettings.json schema, I would add tests. At least the `null` case should be taken into account, and it should throw a `NullReferenceException` at the `Value.ToString()` part at the moment:
```json
{
  "ConnectionStrings": {
    "myDb1": "Server=myServer;Database=myDb1;Trusted_Connection=True;",
    "invalid1": [
      "myServer",
      "myDb1"
    ],
    "invalid2": {
      "Server": "myServer",
      "Database": "myDb1"
    },
    "invalid3": null
  }
}
```

Another problematic json might look like this:
```json
{
  "ConnectionStrings": [
    "Server=myServer;Database=myDb1;Trusted_Connection=True;",
    "Server=myServer;Database=myDb2;Trusted_Connection=True;"
  ]
}
```

Or like this
```json
{
  "ConnectionStrings": "Server=myServer;Database=myDb1;Trusted_Connection=True;"
}
```

Or like this
```json
{
  "ConnectionStrings": null
}
```